File: apps/web/src/entities/week/api/schemas.ts
import { z } from 'zod'

export const WeekMilestoneCreateSchema = z.object({
  sprint_uuid: z.string(),
  owner_user_uuid: z.string(),
  title: z.string().min(1).max(512),
  due_on: z.string().nullable().optional(),
})
export type WeekMilestoneCreate = z.infer<typeof WeekMilestoneCreateSchema>

export const WeekMilestoneUpdateSchema = z.object({
  title: z.string().min(1).max(512).optional(),
  owner_user_uuid: z.string().optional(),
  due_on: z.string().nullable().optional(),
})
export type WeekMilestoneUpdate = z.infer<typeof WeekMilestoneUpdateSchema>

export const WeekMilestoneReadSchema = z.object({
  uuid: z.string(),
  sprint_uuid: z.string(),
  owner_user_uuid: z.string(),
  title: z.string(),
  due_on: z.string().nullable(),
  completed_at: z.string().nullable(),
  created_at: z.string(),
  updated_at: z.string(),
})
export type WeekMilestoneRead = z.infer<typeof WeekMilestoneReadSchema>

File: server/src/app/week/schemas.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WeekMilestoneCreate(BaseModel):
    sprint_uuid: str
    owner_user_uuid: str
    title: str = Field(min_length=1, max_length=512)
    due_on: datetime | None = None


class WeekMilestoneUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=512)
    owner_user_uuid: str | None = None
    due_on: datetime | None = None


class WeekMilestoneRead(BaseModel):
    uuid: str
    sprint_uuid: str
    owner_user_uuid: str
    title: str
    due_on: datetime | None = None
    completed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
